src/coreclr-debug/main.ts

```diff
@@ -1,263 +1,263 @@
+    
+    if (CoreClrDebugUtil.existsSync(_util.installCompleteFilePath())) {
+        console.log('.NET Core Debugger tools already installed');
+        return;
+    }
+    
+    if (!isOnPath('dotnet')) {
+        const getDotNetMessage = "Get .NET CLI tools"; 
+        vscode.window.showErrorMessage("The .NET CLI tools cannot be located. .NET Core debugging will not be enabled. Make sure .NET CLI tools are installed and are on the path.",
+            getDotNetMessage).then(function (value) {
+                if (value === getDotNetMessage) {
+                    let open = require('open');
+                    open("http://dotnet.github.io/getting-started/");
+                }
+            });
+            
+        return;
+    }
+    
+    _reporter = reporter;
+    _channel = vscode.window.createOutputChannel('coreclr-debug');
+    
+    // Create our log file and override _channel.append to also output to the log
+    _installLog = fs.createWriteStream(_util.installLogPath());
+    (function() {
+        let proxied = _channel.append;
+        _channel.append = function(val: string) {
+            _installLog.write(val);
+            proxied.apply(this, arguments);
+        };
+    })();
+
+    let statusBarMessage = vscode.window.setStatusBarMessage("Downloading and configuring the .NET Core Debugger...");
+       
+    let installStage = 'installBegin';
+    let installError = '';
+    
+    writeInstallBeginFile().then(function() {
+        installStage = 'dotnetRestore'
+        spawnChildProcess('dotnet', ['--verbose', 'restore', '--configfile', 'NuGet.config'], _channel, _util.coreClrDebugDir())  
+    }).then(function() {
+        installStage = "dotnetPublish";
+        return spawnChildProcess('dotnet', ['--verbose', 'publish', '-o', _util.debugAdapterDir()], _channel, _util.coreClrDebugDir());
+    }).then(function() {
+        installStage = "ensureAd7";
+        return ensureAd7EngineExists(_channel, _util.debugAdapterDir());
+    }).then(function() {
+        installStage = "additionalTasks";
+        let promises: Promise<void>[] = [];
+
+        promises.push(renameDummyEntrypoint());
+        promises.push(removeLibCoreClrTraceProvider());
+
+        return Promise.all(promises);
+    }).then(function() {
+        installStage = "rewriteManifest";
+        rewriteManifest();
+        installStage = "writeCompletionFile";
+        return writeCompletionFile();
+    }).then(function() {
+        installStage = "completeSuccess";
+        vscode.window.setStatusBarMessage('Successfully installed .NET Core Debugger.');
+    })
+    .catch(function(error) {
+        const viewLogMessage = "View Log";
+        vscode.window.showErrorMessage('Error while installing .NET Core Debugger.', viewLogMessage).then(function (value) {
+            if (value === viewLogMessage) {
+                _channel.show(vscode.ViewColumn.Three);
+            }
+        });
+        statusBarMessage.dispose();
```

Comment:
Are you asking if we want to dispose the first message before setting the new success message? Probably.